### PR TITLE
enhancement(CI): Goreleaser with ksctl version

### PR DESCRIPTION
**Pull request by kranurag78**
- update goreleaser and add version command

### CODE REVIEW

**Comment by dipankardas011** [comment]
@kranurag78 lets add release number to the version 😄

**Comment by dipankardas011** [comment]
@kranurag78 what to do with this PR?

File: cli/cmd/version.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+)
+
+const ksctl  = `
+ _             _   _ 
+| | _____  ___| |_| |
+| |/ / __|/ __| __| |
+|   <\__ \ (__| |_| |
+|_|\_\___/\___|\__|_|
+`
+var Version string = "dev"
```
[the remark this answers is not in the document]
Comment (dipankardas011) [reply cli/cmd/version.go:17]: yes got it

File: .goreleaser.yaml [hunk 2]
```diff
@@ -10,10 +10,9 @@ builds:
     goarch:
       - amd64
       - arm64
-    ldflags: -w -s -X github.com/kubesimplify/ksctl/selfupdate.BuildDate={{.Date}} -X github.com/kubesimplify/ksctl/selfupdate.GitSummary={{.Version}}
+    ldflags: -w -s -X github.com/kubesimplify/ksctl/cli/cmd.BuildDate={{.Date}} -X github.com/kubesimplify/ksctl/cli.cmd.Commit=$(git rev-parse --short HEAD) -X github.com/kubesimplify/ksctl/cli/cmd.Version=$(git describe --tags --abbrev=0)
```
Comment (dipankardas011) [inline .goreleaser.yaml:13]: so you have added the Version tag as well ?

**Review by dipankardas011 — COMMENTED** [review]
Will check and merge

**Comment by kranurag78** [comment]
The main aim of this PR is to add `ldflags` which was implemented wrong previously.

File: .goreleaser.yaml (hunk 2 above)
Comment (kranurag78) [reply .goreleaser.yaml:13]:
```bash
-X github.com/kubesimplify/ksctl/cli/cmd.Version=$(git describe --tags --abbrev=0
```
For version, I'm relying on this but AFAIK this one is not working. This was the reason for draft PR.

Comment (dipankardas011) [reply .goreleaser.yaml:13]: Okay will check and fix this

**Comment by dipankardas011** [comment]
```bash
VERSION=$(git describe --tags --abbrev=0) GITCOMMIT=$(git rev-parse --short HEAD 2> /dev/null || true) DATE=$(date) goreleaser build -f .goreleaser
.yaml --clean
```

**Comment by dipankardas011** [comment]
```yml=goreleaser.yml
builds:
  - binary: ksctl
    main: ./cli
    goos:
      - linux
      # - darwin
      # - windows
    goarch:
      - amd64
      # - arm64
    ldflags: 
      - -s -w -X "github.com/kubesimplify/ksctl/cli/cmd.BuildDate={{.Env.DATE}}"
      - -s -w -X "github.com/kubesimplify/ksctl/cli/cmd.Version={{.Env.VERSION}}"
      - -s -w -X "github.com/kubesimplify/ksctl/cli.cmd.Commit={{.Env.GITCOMMIT}}"
    env:
      - CGO_ENABLED=0
```

**Comment by dipankardas011** [comment]
Still this is showing 
image

**Comment by dipankardas011** [comment]
@kranurag78

**Comment by dipankardas011** [comment]
> Still this is showing image

no commit

**Pull request closed without merging**